packages/@atjson/renderer-commonmark/src/index.ts

```diff
@@ -433,6 +433,15 @@ export default class CommonmarkRenderer {
    */
   *'paragraph'(): Iterable<any> {
     let text = yield;
+
+    // Remove leading and trailing newlines from paragraphs
+    // with text in them.
+    // This ensures that paragraphs preceded with tabs or spaces
+    // will not turn into code blocks
+    if (!text.match(/^\s+$/g)) {
+      text = text.replace(/^\s+/g, '').replace(/\s+$/g, '');
```

Comment:
Should we escape the spaces instead? This is Google Doc's behaviour, since inserting leading space in a wysiwyg environment suggests that it's whitespace that is intended to be preserved.

Reply:
Yeah, this is a bit tricky. I don't think our editors want preceding spaces to be significant in normal environments, and it may be that we allow preformatted text inside of other annotations.

Reply:
Hmm - I'd be interested in where we see this occur; my hunch is pretty strongly that:

1. At the beginnings of lines, whitespace (with perhaps the exception of one or two spaces?) is intended to be significant, since it's very visible if present.
2. In other parts of the text (e.g., inside paragraphs, end of sentences, end of lines) whitespace is not significant -- with the caveat that end-of-sentence spacing isn't very well encoded, but definitely a stylistic preference at the brand-design level, but will vary from editor to editor ("one space or two?")
3. We should definitely also have a "significant whitespace" or "poetry" annotation to make the whitespace intent explicit.

For cases where leading whitespace isn't intended to be significant, the corrective path for editors is to delete the leading whitespace. Where it _is_ intended to be significant, we currently don't give them any way to do that. :(

(I won't die on this hill, but escaping rather than stripping leading whitespace seems an easy hack that gives power to editors while not causing any undue problems for us or consumer devs)